### PR TITLE
fix: passing serve_cmd and passthrough kwargs

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron <29749331+aarnphm@users.noreply.github.com>

## What does this PR address?

This PR addresses two issues:
1. Currently on main, `bentoml.Server` defaults to use `serve-http` as serve_cmd, which means GrpcServer won't work. This PR fixes that by passing `serve_cmd` to the server args.
2. After creating the server, there is no way to pass down subprocess args to the actual process. This PR adds `**kwargs` as passthrough to server.process 

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/server.py [hunk 1]
```diff
@@ -93,14 +93,15 @@ def __init__(
...
-            def __init__(self, blocking: bool = False):
+            def __init__(self, blocking: bool = False, **kwargs: t.Any):
                 logger.info(f"starting server with arguments: {server.args}")
 
                 server.process = subprocess.Popen(
                     server.args,
                     stdout=subprocess.PIPE,
                     stderr=subprocess.PIPE,
                     stdin=subprocess.PIPE,
+                    **kwargs,
```
Comment (sauyon) [inline src/bentoml/server.py:104]: I think we probably should support specific arguments, was there something you thought should be supported?

Maybe we should have something to configure the standard in/outs.

**Review by sauyon — CHANGES_REQUESTED** [review]

File: src/bentoml/server.py [hunk 2]
```diff
@@ -93,14 +93,17 @@ def __init__(
 
     def _create_startmanager(server):  # type: ignore # not calling self self
         class _StartManager:
-            def __init__(self, blocking: bool = False):
+            def __init__(
+                self, blocking: bool = False, env: dict[str, str] | None = None
```
Comment (parano) [inline src/bentoml/server.py:97]: should we add a docstring here? users may confuse this with our env manager parameter

File: src/bentoml/server.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def __init__(
             sys.executable,
             "-m",
             "bentoml",
-            "serve-http",
+            serve_cmd,
```
[the remark this answers is not in the document]
Comment (sauyon) [reply src/bentoml/server.py:67]: This is mostly internal, I think it's fine.

File: src/bentoml/server.py (hunk 2 above)
Comment (aarnphm) [reply src/bentoml/server.py:97]: Done.

**Comment by aarnphm** [comment]
cc @sauyon ready for another round of review

**Review by sauyon — APPROVED** [review]

**Pull request merged**